document drawbacks of renaming widely used package (#886)
ropensci/dev_guide: maintenance_evolution.Rmd

Once a function is made defunct, its tests are simply removed.

## Renaming packages {#renaming-packages}

> Renaming a package that is in very early development is fine.

Reviewer: ```suggestion
Renaming a package that is in early development is fine.
```

It can be the opportunity, before review, to comply with our [naming advice](#naming-your-package).

> Now, renaming a package that has been widely adopted and/or released on CRAN is problematic.

Reviewer: ```suggestion
Renaming a package that is already widely adopted and/or released on CRAN is problematic.
```

CRAN has a [policy](https://cran.r-project.org/web/packages/policies.html) stating that Package names on CRAN are persistent and in general it is not permitted to change a package’s name.

> The package under its old name might be a dependency of packages, scripts, and feature in documentation and blog posts, etc.

Reviewer: ```suggestion
The package under its old name might be a dependency of packages, scripts, and feature in scientific publications, and blog posts, etc.
```

Reviewer 2: We can add "scientific publications" without removing documentation.  But definitely add "scientific publications"

> Creating a newer package that is a rewrite with a better interface, like [httr2 which is the second generation of httr](https://www.tidyverse.org/blog/2023/11/httr2-1-0-0/) ; or creating editions of a package like for [testthat](https://testthat.r-lib.org/articles/third-edition.html), are better strategies.

Reviewer 3: ```suggestion
Instead of renaming, it is better to create a newer package that is a rewrite with a better interface, like [httr2 which is the second generation of httr](https://www.tidyverse.org/blog/2023/11/httr2-1-0-0/) ; or to create editions of a package like for [testthat](https://testthat.r-lib.org/articles/third-edition.html).
```

> Creating a newer package that is a rewrite with a better interface, like [httr2 which is the second generation of httr](https://www.tidyverse.org/blog/2023/11/httr2-1-0-0/) ; or creating editions of a package like for [testthat](https://testthat.r-lib.org/articles/third-edition.html), are better strategies.

Reviewer: ```suggestion
When radically changing the interface, starting a new package from scratch, like [httr2 which is the second generation of httr](https://www.tidyverse.org/blog/2023/11/httr2-1-0-0/) ; or creating editions of a package like for [testthat](https://testthat.r-lib.org/articles/third-edition.html), are better strategies. If you also maintain the old package, you can soft-deprecate it with a startup message, such as in httr.
```

> In particular, they allow users to choose when/whether to update their codebase to the new package or edition.

Reviewer: ```suggestion
This allows users and package authors to choose when/whether to update their codebase to the new package or edition.
```

> In particular, they allow users to choose when/whether to update their codebase to the new package or edition.

Reviewer 3: ```suggestion
Both of these strategies allow users to choose when/whether to update to the new package or edition.
```

> If you rewrite a package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were author of the original package.

Reviewer: ```suggestion
If you copy code from another package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were author of the original package.
```

> If you rewrite a package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were author of the original package.

Reviewer 3: ```suggestion
If you rewrite a package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were authors of the original package.
```

> If you copy code from another package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were author of the original package.

Reviewer 3: ```suggestion
If you copy code from another package, make sure to acknowledge authors of code you re-use by listing them in DESCRIPTION with a comment that states they were authors of the original package.
```
